amr-wind/diffusion/incflo_diffusion.cpp

```diff
@@ -197,4 +197,69 @@ void fixup_eta_on_domain_faces(
         }
     }
 }
+
+void viscosity_to_uniform_space(
+    amrex::Array<amrex::MultiFab, AMREX_SPACEDIM>& b,
+    const amr_wind::FieldRepo& repo,
+    int lev)
+{
+    const auto& mesh_fac_xf =
+        repo.get_mesh_mapping_field(amr_wind::FieldLoc::XFACE);
+    const auto& mesh_fac_yf =
+        repo.get_mesh_mapping_field(amr_wind::FieldLoc::YFACE);
+    const auto& mesh_fac_zf =
+        repo.get_mesh_mapping_field(amr_wind::FieldLoc::ZFACE);
+    const auto& mesh_detJ_xf =
+        repo.get_mesh_mapping_detJ(amr_wind::FieldLoc::XFACE);
+    const auto& mesh_detJ_yf =
+        repo.get_mesh_mapping_detJ(amr_wind::FieldLoc::YFACE);
+    const auto& mesh_detJ_zf =
+        repo.get_mesh_mapping_detJ(amr_wind::FieldLoc::ZFACE);
+
+    // TODO: should box loops below be over grown tile box ?
```

Comment:
have you tried running the same case with varying block size and mpi ranks? that might help you figure this out? or delete comment if it's no longer a mystery